### PR TITLE
docs(installation) Databases installation in kubernetes. Fix #41.

### LINKED ISSUE #41: Document how to configure the database for Harbor

### PR DESCRIPTION
**Pull request by holyhope**
Step by step explanation to deploy postgresql databases in kubernetes with Helm
and the bitnami chart.

Signed-off-by: Pierre Péronnet <pierre.peronnet@ovhcloud.com>

### CODE REVIEW

File: docs/installation/redis-installation.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
+# Redis installation
+
+Many application of the Harbor stack require a Postgresql database.
```
Comment (steven-zou) [inline docs/installation/redis-installation.md:3]:
```suggestion
Many application of the Harbor stack require a Redis database.
```
?

File: docs/installation/redis-installation.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+## With helm
+
+Recommended: Install the [postgresql helm chart](https://github.com/bitnami/charts/tree/master/bitnami/postgresql) from bitnami repository.
```
Comment (steven-zou) [inline docs/installation/redis-installation.md:7]:
```suggestion
Recommended: Install the [postgresql helm chart](https://github.com/bitnami/charts/tree/master/bitnami/redis-ha) from bitnami repository.
```
?

**Comment by bitsf** [comment]
what's the difference between dandydev/redis-ha and stable/redis-ha

**Comment by holyhope** [comment]
> what's the difference between dandydev/redis-ha and stable/redis-ha

`stable/redis-ha` is now deprecated in favor of `dandydev/redis-ha`. See [deprecation warning](https://github.com/helm/charts/tree/master/stable/redis-ha#deprecation-warning).

File: docs/installation/redis-installation.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
+# Redis installation
+
+Many application of the Harbor stack require a Redis database.
+
+## With helm
+
+Recommended: Install the [postgresql helm chart](https://github.com/bitnami/charts/tree/master/bitnami/redis-ha) from bitnami repository.
```
Comment (renmaosheng) [inline docs/installation/redis-installation.md:7]: should be redis?

File: docs/installation/redis-installation.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+   ```bash
+   export B64_PASSWORD="$(kubectl get secret "$COMPONENT-redis" -o "jsonpath={.data['auth']}")"
+   ```
+
+3. Create the computed secret with correct keys (see [`api/v1alpha1/harbor_secret_format.go`](../../api/v1alpha1/harbor_secret_format.go))
+
+   ```bash
+   # url: redis://registry-cache-redis-ha-announce-0:6379/0
+   # namespace: ''
+   kubectl create secret "$COMPONENT-database" \
```
Comment (renmaosheng) [inline docs/installation/redis-installation.md:42]: should be redis?

File: docs/installation/redis-installation.md (hunk 3 above)
Comment (renmaosheng) [inline docs/installation/redis-installation.md:7]: should be redis

File: docs/installation/installation.md [hunk 5]
```diff
@@ -33,7 +33,9 @@ Kubernetes API running (see [Supported platforms](https://github.com/goharbor/ha
    make docker-push IMG=the.registry/goharbor/harbor-operator:dev
    ```
 
-3. Deploy requirements
+3. Deploy requirements.
+   The following command deploys [databases](./database-installation.md)
+   and [redis](./redis-installation.md) needed to run an harbor.
```
Comment (renmaosheng) [inline docs/installation/installation.md:38]: an -> 'a'

**Comment by bitsf** [comment]
How about using `helm install registry-cache bitnami/redis --set sentinel.enabled=true --set usePassword=false` , that we can only add one bitnami repo for helm

**Comment by steven-zou** [comment]
> How about using `helm install registry-cache bitnami/redis --set sentinel.enabled=true --set usePassword=false` , that we can only add one bitnami repo for helm

@holyhope 

Any comments for zm's suggestion?

**Comment by holyhope** [comment]
> How about using `helm install registry-cache bitnami/redis --set sentinel.enabled=true --set usePassword=false`, that we can only add one bitnami repo for helm

Totally agree about bitnami, updating the PR...
About the `sentinel.enabled` since sentinel is not yet working with Harbor 1.10, I suggest that we create an issue about this to fix it later. WDYT?
Not sure why disabling password. IMO it is better to have password on all services.

**Comment by holyhope** [comment]
FYI tests are fixed with that: https://github.com/goharbor/harbor-operator/pull/49

**Pull request merged**